File: pathplannerlib/src/test/java/com/pathplanner/lib/path/GoalEndStateTest.java
package com.pathplanner.lib.path;

import static org.junit.jupiter.api.Assertions.assertEquals;

import edu.wpi.first.math.geometry.Rotation2d;
import org.json.simple.JSONObject;
import org.junit.jupiter.api.Test;

public class GoalEndStateTest {
  public static final double DELTA = 1e-3;

  @Test
  public void testGetters() {
    GoalEndState endState = new GoalEndState(2.0, Rotation2d.fromDegrees(35));

    assertEquals(2.0, endState.getVelocity(), DELTA);
    assertEquals(Rotation2d.fromDegrees(35), endState.getRotation());
  }

  @Test
  public void testFromJson() {
    JSONObject json = new JSONObject();
    json.put("velocity", 1.25);
    json.put("rotation", -15.5);

    assertEquals(
        new GoalEndState(1.25, Rotation2d.fromDegrees(-15.5)), GoalEndState.fromJson(json));
  }
}

File: pathplannerlib/src/test/native/cpp/pathplanner/lib/path/GoalEndStateTest.cpp
#include <gtest/gtest.h>
#include "pathplanner/lib/path/GoalEndState.h"

using namespace pathplanner;

TEST(GoalEndStateTest, TestGetters) {
	GoalEndState endState(2_mps, wpi::math::Rotation2d(35_deg));

	EXPECT_DOUBLE_EQ(2.0, endState.getVelocity()());
	EXPECT_EQ(wpi::math::Rotation2d(35_deg), endState.getRotation());
}

TEST(GoalEndStateTest, TestFromJson) {
	wpi::util::json json = wpi::util::json::object("velocity", 1.25,
			"rotation", -15.5);

	EXPECT_EQ(GoalEndState(1.25_mps, wpi::math::Rotation2d(-15.5_deg)), GoalEndState::fromJson(json));
}
